src/cosim/timer.cpp

```diff
@@ -18,11 +16,55 @@ constexpr std::chrono::microseconds MIN_SLEEP(100);
 namespace cosim
 {
 
+void real_time_config::set_real_time_simulation(bool enable)
+{
+    if (realTimeSimulation_ != enable) {
+        configChanged_.store(true);
```

Comment:
Even though the individual variables are atomic, you could end up in a situation like the following, where thread A wants to change the configuration and thread B wants to read it:

1. Thread A sets `configChanged_` to `true`.
2. Thread B sees that `configChanged_` is `true`, reads `realTimeSimulation_`.
3. Thread A sets `realTimeSimulation_` to new value.
4. Thread B sets `configChanged_` to `false`.

In other words, B gets the old value even though it deduced that the configuration had changed, and at the end of it all, the configuration is marked ad "not changed" even though it is.

I'm not 100% sure whether that's a problem here, but maybe you can tell?

Reply:
You are right! The consequences shouldn't be *terrible*, but it means I overthought/engineered this part and can probably simplify it. Will update!